src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -346,6 +347,45 @@ worker_partial_agg_sfunc(PG_FUNCTION_ARGS)
 	{
 		box = pallocInAggContext(fcinfo, sizeof(StypeBox));
 		box->agg = PG_GETARG_OID(1);
+
+		/* must verify argument type matches supplied aggregate's argument type */
+		Form_pg_proc procform;
```

Comment:
Maybe move this whole block of new code to a separate function, to keep the signature validation clearly separated from the actual logic.